src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -519,18 +521,29 @@ UCS_CLASS_INIT_FUNC(uct_rc_mlx5_iface_common_t,
     self->super.rx.srq.quota       = self->rx.srq.mask + 1;
     self->super.config.exp_backoff = mlx5_config->exp_backoff;
 
-    switch (self->super.config.fence_mode) {
+    switch (rc_config->fence_mode) {
```

Comment:
there is some code duplication here
in order to fix it:
```
if (fence_mode == strong) {
  //setup strong
} else if ((fence_mode=weak) || ( (fence_mode==auto) && has_pci_atomics)  {
  // setup weak
} else {
 // setup none
}

Reply:
ok, refactored